### ISSUE TITLE
ddr sync not pushing commits to mits remote correctly

### ISSUE THREAD

**Issue by GeoffFroh** [densho/ddr-cmdln#16]
ddr sync is pushing local changes to synced/master on mits; but not to master as expected. 

Test routine was as follows:

1. I made a simple change to text of collection.json, then manually committed.
```
ddr@kinkura:/ddr-testing-247$ nano collection.json
ddr@kinkura:/ddr-testing-247$ git status
# On branch master
# Changes not staged for commit:
#   (use "git add <file>..." to update what will be committed)
#   (use "git checkout -- <file>..." to discard changes in working directory)
#
#       modified:   collection.json
#
no changes added to commit (use "git add" and/or "git commit -a")
ddr@kinkura:/ddr-testing-247$ git add collection.json
ddr@kinkura:/ddr-testing-247$ git commit -m"Manual commit of test changes for ddr sync testing."
[master 9e16e03] Manual commit of test changes for ddr sync testing.
 1 file changed, 1 insertion(+), 1 deletion(-)
```

2. git status shows as one commit ahead of remote.
```
ddr@kinkura:/ddr-testing-247$ git status
# On branch master
# Your branch is ahead of 'origin/master' by 1 commit.
#
nothing to commit (working directory clean)
```

3. Ran ddr sync
```
ddr@kinkura:/usr/local/src/ddr-local/ddrlocal$ ddr sync -u DDRAdmin -m kinkura@hq.densho.org -c /ddr-testing-247
```

4. Went back to local repo dir and checked status. Shows one commit ahead still. I tried ddr sync'ing twice for good measure with the same result. cgit on mits shows synced/master branch ahead of master.
```
ddr@kinkura:/usr/local/src/ddr-local/ddrlocal$ cd /ddr-testing-247
ddr@kinkura:/ddr-testing-247$ git status
# On branch master
# Your branch is ahead of 'origin/master' by 1 commit.
#
nothing to commit (working directory clean)
```

5. Then tried a manual push from local. The commit is pushed to master on mits and cgit shows synced/master and master are the same.
```
ddr@kinkura:/ddr-testing-247$ git push
Total 0 (delta 0), reused 0 (delta 0)
To git@mits.densho.org:ddr-testing-247.git
   6b915bd..9e16e03  master -> master
ddr@kinkura:/ddr-testing-247$ git status
# On branch master
nothing to commit (working directory clean)
```

Notes:
- Script returns no error.
- Running 'git push' manually from local will push changes to remote master.

**Comment by gjost** [comment]
Note: Use `git annex sync $REMOTE` instead of `git push $REMOTE`.
Reason: `git-push` only pushes the current branch to the remote.  Git-annex has to synchronize multiple branches -- the current branch (usually `master`) plus `annex` -- so `git-push` is not enough.

**Comment by gjost** [comment]
Commits *are* being pushed to the remote, but they are not being merged into master on that remote.

**Comment by gjost** [comment]
Hub server "mits" is running
* git-annex 3.20120629~bpo60+2
* git 1.7.2.5
* gitolite3 v3.2-19-gb9bbb78

Local machine is running
* git-annex 3.20120629
* git 1.7.10.4

**Comment by gjost** [comment]
Something is interrupting the sync process before `synced/master` can be merged into `master` on the hub server (mits).  I trace this to commit 1857a7aa3f (master), in which I replaced the manual sync code with the official `git annex sync`.

IIRC I did the manual sync in the first place because I feared that `git annex sync` might be doing operations I did not want, such as `git commit -a`.  The git-annex assistant was doing these in order to make for a simple Dropbox-like user experience.

Commit 1857a7aa3f was intended to reduce the amount of custom code by just using `git-annex`.  I looks like that was premature, so the manual code has been restored, with slightly clearer logging and documentation.
